ZenPacks/training/OpenWeather/dsplugins.py

```diff
@@ -65,78 +62,86 @@ def collect(self, config):
-            except (KeyError, TypeError):
-                LOG.exception(
-                    "%s: failed to get conditions data for %s",
-                    config.id,
-                    datasource.params['location_name'])
+            except (KeyError, IndexError), e:
+                message = "{}: failed to parse conditions data for {}" \
+                    .format(config.id, datasource.params['location_name'])
+
+                LOG.exception(message)
 
                 data['events'].append({
                     'device': config.id,
                     'component': datasource.component,
-                    'eventKey': 'oweather-conditions',
-                    'eventClassKey': 'oweather',
+                    'eventKey': 'oweatherConditionsParse_%s' % datasource.params['location_name'],
+                    'eventClassKey': 'oweatherConditions',
                     'severity': SEVERITY_ERROR,
-                    'message':
-                        '{}: failed to get conditions data for {}'
-                            .format(config.id, datasource.params['location_name']),
+                    'summary': message,
+                    'message': e
                 })
 
                 returnValue(data)
+            else:
+                data['events'].append({
+                    'device': config.id,
+                    'component': datasource.component,
+                    'eventKey': 'oweatherConditionsParse_%s' % datasource.params['location_name'],
+                    'eventClassKey': 'oweatherConditions',
+                    'severity': SEVERITY_CLEAR,
+                    'summary': '',
+                })
 
-            for datapoint_id in (x.id for x in datasource.points):
-                json_key = self.camel_case_to_snake(datapoint_id)
+            for datapointId in (x.id for x in datasource.points):
+                jsonKey = camelCaseToSnake(datapointId)
                 for block in (weather, main, wind):
-                    if json_key not in block:
+                    if jsonKey not in block:
                         continue
-
                     try:
-                        value = block[json_key]
-                        if isinstance(value, basestring):
-                            value = value.strip(' %')
-
-                        value = float(value)
+                        value = float(block[jsonKey])
 
-                        if datapoint_id == 'pressure':
+                        if datapointId == 'pressure':
                             value *= 100
 
                     except (TypeError, ValueError):
                         # Sometimes values are NA or not available.
                         continue
 
-                    dpname = '_'.join((datasource.datasource, datapoint_id))
+                    dpname = '_'.join((datasource.datasource, datapointId))
                     data['values'][datasource.component][dpname] = (value, 'N')
 
             data['maps'].append(
```

Comment:
keep in mind in real world it's not the best idea to follow :) 
take a look at datasource property of components fileds
https://zenpack-sdk.zenoss.com/en/latest/yaml-classes-and-relationships.html#class-property-fields